packages/common-server/src/filesv2.ts

```diff
@@ -453,4 +453,60 @@ export function dot2Slash(fname: string) {
   return fname.replace(/\./g, "/");
 }
 
+/** Checks that the `path` contains a file. */
+export async function fileExists(path: string) {
+  try {
+    const stat = await fs.stat(path);
+    return stat.isFile();
+  } catch {
+    return false;
+  }
+}
+
+/** Finds if a file `fpath` is located in any vault.
+ *
+ * @param fpath A file name or relative path that we are searching inside vaults.
+ */
+async function findFileVault({
+  fpath,
+  wsRoot,
+  vaults,
+}: {
+  fpath: string;
+  wsRoot: string;
+  vaults: DVault[];
+}): Promise<{ vault: DVault; fullPath: string } | undefined> {
+  // Assets from later vaults will overwrite earlier ones.
+  vaults = [...vaults].reverse();
+  for (const vault of vaults) {
+    const fullPath = path.join(wsRoot, vault.fsPath, fpath);
```

Comment:
heads up that the `fullPath` changes depending if we have a workspace vault or seed vault. see https://github.com/dendronhq/dendron-docs/blob/main/vault/pkg.dendron-engine.dev.cook.md#getting-absolute-path-for-a-vault